data-prepper-plugins/otel-trace-raw-processor/src/main/java/com/amazon/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.oteltrace;
+
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.processor.AbstractProcessor;
+import com.amazon.dataprepper.model.processor.Processor;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.model.trace.Span;
+import com.amazon.dataprepper.plugins.processor.oteltrace.model.SpanSet;
+import com.amazon.dataprepper.plugins.processor.oteltrace.model.TraceGroup;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.micrometer.core.instrument.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+
+
+@DataPrepperPlugin(name = "otel_trace_raw", pluginType = Processor.class)
+public class OTelTraceRawProcessor extends AbstractProcessor<Record<Span>, Record<Span>> {
+    private static final long SEC_TO_MILLIS = 1_000L;
+    private static final Logger LOG = LoggerFactory.getLogger(OTelTraceRawProcessor.class);
+
+    private final long traceFlushInterval;
+
+    private final Map<String, SpanSet> traceIdSpanSetMap = new ConcurrentHashMap<>();
+
+    private final Cache<String, TraceGroup> traceIdTraceGroupCache;
+
+    private long lastTraceFlushTime = 0L;
+
+    private final ReentrantLock traceFlushLock = new ReentrantLock();
+    private final ReentrantLock prepareForShutdownLock = new ReentrantLock();
+
+    private volatile boolean isShuttingDown = false;
+
+    public OTelTraceRawProcessor(final PluginSetting pluginSetting) {
+        super(pluginSetting);
+        traceFlushInterval = SEC_TO_MILLIS * pluginSetting.getLongOrDefault(
+                OtelTraceRawProcessorConfig.TRACE_FLUSH_INTERVAL, OtelTraceRawProcessorConfig.DEFAULT_TG_FLUSH_INTERVAL_SEC);
+        final int numProcessWorkers = pluginSetting.getNumberOfProcessWorkers();
+        traceIdTraceGroupCache = CacheBuilder.newBuilder()
+                .concurrencyLevel(numProcessWorkers)
+                .maximumSize(OtelTraceRawProcessorConfig.MAX_TRACE_ID_CACHE_SIZE)
+                .expireAfterWrite(OtelTraceRawProcessorConfig.DEFAULT_TRACE_ID_TTL_SEC, TimeUnit.SECONDS)
+                .build();
+    }
+
+    /**
+     * execute the prepper logic which could potentially modify the incoming record. The level to which the record has
```

Comment:
Please update the javadocs. The current form is a duplicate of the interface and references "prepper"